WcaOnRails/app/models/competition.rb

```diff
@@ -424,7 +425,7 @@ def belongs_to_region?(region)
   end
 
   def contains?(search_param)
-    [name, cityName, venue, cellName, countryId, start_date.strftime('%B')].any? do |field|
+    [name, cityName, delegates.map(&:name).join(','), venue, cellName, countryId, start_date.strftime('%B')].any? do |field|
```

Comment:
`delegates.pluck(:name).join(',')` should be more effective as it only retrieves names instead of the whole delegates from db.
Also it could be more flexible and allow searching for '<city> <delegate1> <delegate2>' and such. I can imagine this like so:
```ruby
search_param.split.all? do |part|
  [<fields>].any? do |field|
    ...
  end
end
```
Worth testing.